runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/StreamingEngineClient.java

```diff
@@ -226,8 +278,7 @@ private void startWorkerMetadataConsumer() {
         });
   }
 
-  @VisibleForTesting
-  void finish() {
+  public void finish() {
     if (!started.compareAndSet(true, false)) {
```

Comment:
similarly could make it an error if not started
and if we don't need to support starting/stopping/restarting don't set back to false

Reply:
done